### PR TITLE
Replace file_get_contents by curl_exec

### PR DESCRIPTION
**Pull request by korentzari**
Replace file_get_contents by curl_exec. 
Solve possible problem cloudflare

### CODE REVIEW

**Comment by Chefkeks** [comment]
Move the function to a logic file please. We may can use them somewhere else too.

**Comment by korentzari** [comment]
Move and dedicated logic file. And use with getZeCharles.php

**Comment by Chefkeks** [comment]
Seems fine to me. Can you take a quick final look @Artanicus before the merge?

**Comment by Artanicus** [comment]
Seems fine but raises a few questions:
- Why does file_get_contents fail when curl succeeds? Something to do with ipv6 being badly implemented in either PHP or on bot admin's systems.
- We use file_get_contents in other places as well but those are files, not URLs. getDB.php also uses it but since that's run inoften and not by bot admins that should be fine.

TL;DR; yeah, let's merge it.
